### ISSUE TITLE
Unauthorized with IncorrectNonce. authNonce is always empty with silentRenew=true

### ISSUE THREAD

**Issue by zeinaa** [damienbod/angular-auth-oidc-client#576]
After upgrading the library from **8.0.3**  to the latest version I started having this issue. `this.oidcSecurityCommon.authNonce` is always empty and being compared to a successfully decoded id_token nonce. 

![image](https://user-images.githubusercontent.com/20439707/73843351-29c62f80-4827-11ea-8cf4-c9aef25d4725.png)

Here's what I have.
My application has an `unauthorized` page that checks  `IsAuthorized()` and calls `authorize()` if false. Otherwise, it navigates to the main page.

After a successful call to `authorize()`, I redirect to a `callback` page that does the following:
- calls authorizedImplicitFlowCallback : 
```
this.oidcSecurityService.getIsModuleSetup().pipe(
                filter((isModuleSetup: boolean) => isModuleSetup),
                take(1)
            ).subscribe(() => {
                if (window.location.hash)
                {
                    this.oidcSecurityService.authorizedImplicitFlowCallback();
                }
```
- subscribes to onAuthorizationResult to redirect to the main application page:
``` 
this.oidcSecurityService.onAuthorizationResult.pipe(take(1)).subscribe(
                (authResult: AuthorizationResult) => {
               if( authResult.authorizationState === AuthorizationState.authorized) 
               {
                     setTimeout(() => {
                    // Navigate to "main" page - AuthorizationGuard will take care of everything else
                    this.router.navigate(['/-']);
                    }); 
                    }
               } 
```
Additionally, there is a guard added to protected routes that checks for `IsAuthorized()`  and navigates to `unauthorized` if false.

These are my config values for now:
```
{
...
"response_type": "id_token token",
   "start_checksession": false,
    "silent_renew": true,
    "silent_renew_url": "<silent_renew_url>",    
    "silent_renew_offset_in_seconds" : 5,
    "log_console_warning_active": true,
    "log_console_debug_active": true,
    "max_id_token_iat_offset_allowed_in_seconds": 180,
    "iss_validation_off": false,
    "disable_iat_offset_validation" : false,
    "auto_userinfo" : true ,
    "ignore_nonce_after_refresh" : false
}
```

I am still using Implicit flow with silent_renew set to true. It works fine if silent-renew is false.
Any idea if I'm doing something wrong or what I need to be checking?

**Comment by gsingh-dtst** [comment]
@damienbod i am facing the same issue.
any suggestion or any update about it?

**Comment by zeinaa** [comment]
I updated to version 10.0.15 hoping that the subscribe startRenew after isAuthorized is true would solve the problem. It didn't so I changed my code to use CodeFlow with PKCE. I faced an issue again when the silent_renew is true. The code verifier was being sent as empty when getting the token and I was getting the error invalid_grant from the server. It works fine when silent_renew is false.
So I changed my code to set the silent_renew to false on initial setup and after authorization is successful I call setup module again. It is working this way.

```
this.oidcSecurityService.onAuthorizationResult.pipe(take(1)).subscribe(
    (authResult: AuthorizationResult) => {

   if( authResult.authorizationState === AuthorizationState.authorized) 
   {  
        const config: OpenIdConfiguration = {
         silent_renew: true
        };
    this.oidcSecurityService.setupModule(config, authWellknownEndpoints);
}
```
